Source/Core/DolphinQt/Config/ToolTipControls/ToolTipCheckBox.h

```diff
@@ -0,0 +1,28 @@
+// Copyright 2020 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <optional>
+
+#include <QCheckBox>
+#include <QString>
+
+class ToolTipCheckBox : public QCheckBox
```

Comment:
All those classes seem to be extremely similar... You should probably turn this into a class template (`ToolTipWidget<T>?`) instead of having 5 classes with duplicated code.

Some of the classes have extra logic in timerEvent. I'd recommend adding a `onTimerEvent` protected member function that can be overridden by derived classes so you don't need to override timerEvent() and duplicate code. onTimerEvent would be called from ToolTipWidget's timerEvent and would do nothing by default.

Reply:
I did start by looking into a template class but I can't remember why I chose not to do that, I vaguely recall switching designs toward the end so maybe that's why.  I'll give it another whirl, thanks!

Reply:
I did it a little differently than suggested, let me know if you feel like there's a better solution